pkg/certificate/authorization/endpoint.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package authorization
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// ErrEndpoint is the default error class for the authorization endpoint.
+var ErrEndpoint = errs.Class("authorization endpoint error")
+
+// Endpoint implements pb.AuthorizationsServer
+type Endpoint struct {
+	log      *zap.Logger
+	db       *DB
+	server   http.Server
+	listener net.Listener
+}
+
+// NewEndpoint creates a new authorization gRPC server.
+func NewEndpoint(log *zap.Logger, db *DB, listener net.Listener) *Endpoint {
+	mux := http.NewServeMux()
+	endpoint := &Endpoint{
+		log:      log,
+		db:       db,
+		listener: listener,
+		server: http.Server{
+			Handler: mux,
+		},
+	}
+
+	mux.HandleFunc("/v1/authorization/create", endpoint.httpCreate)
+
+	return endpoint
+}
+
+// Create creates an authorization from the given authorization request.
+func (endpoint *Endpoint) Create(ctx context.Context, req *pb.AuthorizationRequest) (_ *pb.AuthorizationResponse, err error) {
+	mon.Task()(&ctx, req.UserId)(&err)
+
+	group, err := endpoint.db.Create(ctx, req.UserId, 1)
```

Comment:
We should add a check to see if there are open auth tokens and return them first, instead of blind generating a new one.
this ensures the users setup a new node (or at least identity) before generating them a new token.